ovh/types_cloud_project_database.go

```diff
@@ -1995,3 +1995,177 @@ func (v CloudProjectDatabaseKafkaUserAccessResponse) ToMap() map[string]interfac
+	Name       string                                                   `json:"name"`
+	Port       int64                                                    `json:"port"`
+	Size       int                                                      `json:"size"`
+	SslMode    *CloudProjectDatabasePostgresqlConnectionPoolSslModeEnum `json:"sslMode,omitempty"`
+	Uri        string                                                   `json:"uri"`
+	UserId     *string                                                  `json:"userId,omitempty"`
+}
+
+func (v CloudProjectDatabasePostgresqlConnectionPoolResponse) ToMap() map[string]interface{} {
+	obj := make(map[string]interface{})
+
+	obj["database_id"] = v.DatabaseId
+	obj["id"] = v.Id
+	obj["mode"] = v.Mode
+	obj["name"] = v.Name
+	obj["port"] = v.Port
+	obj["size"] = v.Size
+	obj["ssl_mode"] = v.SslMode
+	obj["uri"] = v.Uri
+	obj["user_id"] = v.UserId
+
+	return obj
+}
+
+func importCloudProjectDatabasePostgresqlConnectionPool(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	givenId := d.Id()
+	n := 3
+	splitId := strings.SplitN(givenId, "/", n)
+	if len(splitId) != n {
+		return nil, fmt.Errorf("Import Id is not service_name/cluster_id/id formatted")
+	}
+	serviceName := splitId[0]
+	clusterId := splitId[1]
+	id := splitId[2]
+	d.SetId(id)
+	d.Set("cluster_id", clusterId)
+	d.Set("service_name", serviceName)
+
+	results := make([]*schema.ResourceData, 1)
+	results[0] = d
+	return results, nil
+}
+
+func postCloudProjectDatabasePostgresqlConnectionPool(ctx context.Context, d *schema.ResourceData, meta interface{}, dsReadFunc, readFunc schema.ReadContextFunc, updateFunc schema.UpdateContextFunc, f func() interface{}) diag.Diagnostics {
+	serviceName := d.Get("service_name").(string)
+	clusterId := d.Get("cluster_id").(string)
+
+	endpoint := fmt.Sprintf("/cloud/project/%s/database/%s/%s/connectionPool",
+		url.PathEscape(serviceName),
+		url.PathEscape("postgresql"),
+		url.PathEscape(clusterId),
+	)
+
+	params := f()
+	res := &CloudProjectDatabasePostgresqlConnectionPoolResponse{}
+
+	log.Printf("[DEBUG] Will create connectionPool: %+v for cluster %s from project %s", params, clusterId, serviceName)
+	config := meta.(*Config)
+
+	err := config.OVHClient.Post(endpoint, params, res)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(res.Id)
+	return readFunc(ctx, d, meta)
+}
+
+func updateCloudProjectDatabasePostgresqlConnectionPool(ctx context.Context, d *schema.ResourceData, meta interface{}, readFunc schema.ReadContextFunc, f func() interface{}) diag.Diagnostics {
+	config := meta.(*Config)
+	serviceName := d.Get("service_name").(string)
+	clusterId := d.Get("cluster_id").(string)
+	id := d.Id()
+
+	endpoint := fmt.Sprintf("/cloud/project/%s/database/%s/%s/connectionPool/%s",
+		url.PathEscape(serviceName),
+		url.PathEscape("postgresql"),
+		url.PathEscape(clusterId),
+		url.PathEscape(id),
+	)
+
+	params := f()
```

Comment:
```suggestion
	params := (&CloudProjectDatabasePostgresqlConnectionPoolUpdateOpts{}).FromResource(d)
```
As `postCloudProjectDatabasePostgresqlConnectionPool` no need of the `f` parameter